src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -163,8 +163,15 @@ public void start() throws MessageClientException {
     }
 
     private void disconnect() {
-        // 0ms quiescence time, just send the disconnect packet immediately
-        disconnect(0);
+        if (!mqttClientInternal.isConnected()) {
+            return;
+        }
+        try {
+            // 0ms quiescence time, don't wait for DISCONNECT
```

Comment:
What's the 1 for?

Reply:
<= 0 in paho means wait forever 🙃

Reply:
https://github.com/eclipse/paho.mqtt.java/blob/f4e0db802a4433645ef011e711646a09ec9fae89/org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/Token.java#L142-L143

Reply:
so was just picking the next closest thing to "don't wait for disconnect packet"

Reply:
`MqttClient` has a method that doesn't send out disconnect packet, although it's not in `IMqttClient`

Reply:
I mean to explain it in the comment. comment says 0ms, but doesn't mention the 1

Reply:
ahh yeah i added a comment